### PR TITLE
tools: Add genplugin.sh

### PR DESCRIPTION
**Pull request by jd28**
This is a simple bash script for bootstrapping new 2.8 plugins.

### CODE REVIEW

File: tools/genplugin.sh [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+cat > "plugins/$lower/plugin-$lower.cpp" <<-EOT
+#include "NWNX$pname.h"
+
+CNWNX$pname $lower;
+PLUGINLINK *pluginLink = 0;
+
+PLUGININFO pluginInfo = {
+    sizeof(PLUGININFO),
+    "NWNX $pname",
+    PLUGIN_MAKE_VERSION(1, 3, 3, 0),
```
Comment (virusman) [inline tools/genplugin.sh]: Defaults to v. 1.3.3

**Comment by niv** [comment]
Could we not just get rid of all those pluginlink fields? Is anyone seriously using them?

As it stands now, noone cares about the author, licence or url embedded in the binary, and noone maintains proper version numbers. That stuff belongs into README.

**Comment by virusman** [comment]
The version field was originally added to have an ability to check whether a plugin exists and get its version (i.e. inside an NWScript system that relies on an NWNX plugin).

**Comment by niv** [comment]
Yeah, I'm aware. The thing is though, it's buried somewhere and optional, and noone sets it.

I'd propose a small refactor to get rid of all the chaff, and maybe just leave the version in. Then, as soon as I manage to PR my connect stuff, expose plugin versions to the server.

**Comment by virusman** [comment]
The rest of the metadata fields - yeah, they're not really useful at all.

**Comment by virusman** [comment]
Changing common plugin API structs will break all of the out-of-tree plugins out there. If the core is going to be rewritten anyway, I think it'd be nicer to do all of the breaking changes at once, or, alternatively, make a development branch and do all of the core refactoring there.

**Comment by jd28** [comment]
Ok, I've updated this to default to version 1.0.0.  Also updated the usage to indicate that PascalCase is expected as the parameter and to convert that to dash-cash and snake_case where those seem to be the common convention.

**Comment by jd28** [comment]
No point in this being here anymore.

**Pull request closed without merging**